akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala

```diff
@@ -277,6 +279,120 @@ final class MergePreferred[T](val secondaryPorts: Int, val eagerComplete: Boolea
+ *
+ * '''Emits when''' one of the inputs has an element available, preferring
+ * a input based on its priority if multiple have elements available
+ *
+ * '''Backpressures when''' downstream backpressures
+ *
+ * '''Completes when''' all upstreams complete (eagerComplete=false) or one upstream completes (eagerComplete=true), default value is `false`
+ *
+ * '''Cancels when''' downstream cancels
+ *
+ * A `Broadcast` has one `in` port and 2 or more `out` ports.
+ */
+final class MergePrioritized[T] private (val inputPorts: Int, val priorities: Seq[Int], val eagerComplete: Boolean) extends GraphStage[UniformFanInShape[T, T]] {
+  require(inputPorts >= 1, "A Merge must have one or more input ports")
+  require(inputPorts == priorities.size, "Priorities of all inputPorts should be defined")
+
+  val in: immutable.IndexedSeq[Inlet[T]] = Vector.tabulate(inputPorts)(i ⇒ Inlet[T]("MergePrioritized.in" + i))
+  val out: Outlet[T] = Outlet[T]("MergePrioritized.out")
+  override def initialAttributes: Attributes = DefaultAttributes.mergePrioritized
+  override val shape: UniformFanInShape[T, T] = UniformFanInShape(out, in: _*)
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new GraphStageLogic(shape) with OutHandler {
+
+    private val allBuffers = Vector.tabulate(priorities.size)(i ⇒ FixedSizeBuffer[Inlet[T]](priorities(i)))
+    private def pending(buffers: FixedSizeBuffer[Inlet[T]]*): Boolean = buffers.exists(_.nonEmpty)
+
+    private var runningUpstreams = inputPorts
+    private def upstreamsClosed = runningUpstreams == 0
+
+    override def preStart(): Unit = in.foreach(tryPull)
+
+    private def dequeueAndDispatch(): Unit = {
+      toss match {
+        case Some(buffer) ⇒
+          val in = buffer.dequeue()
+          push(out, grab(in))
+          if (upstreamsClosed && !pending(allBuffers: _*)) completeStage() else tryPull(in)
+        case None ⇒
+          if (upstreamsClosed && !pending(allBuffers: _*)) completeStage()
+      }
+    }
+
+    (in zip allBuffers).foreach {
+      case (inlet, buffer) ⇒
+        setHandler(inlet, new InHandler {
+          override def onPush(): Unit = {
+            if (isAvailable(out) && !pending(allBuffers: _*)) {
+              push(out, grab(inlet))
+              tryPull(inlet)
+            } else {
+              buffer.enqueue(inlet)
+            }
+          }
+
+          override def onUpstreamFinish(): Unit = {
+            if (eagerComplete) {
+              in.foreach(cancel)
+              runningUpstreams = 0
+              if (!pending(allBuffers: _*)) completeStage()
+            } else {
+              runningUpstreams -= 1
+              if (upstreamsClosed && !pending(allBuffers: _*)) completeStage()
+            }
+          }
+        })
+    }
+
+    override def onPull(): Unit = {
+      if (pending(allBuffers: _*)) dequeueAndDispatch()
+    }
+
+    setHandler(out, this)
+
+    private def toss = {
```

Comment:
`selectNextElement()`, not a simple accessor so parenthesises would be good to make that clear, also it's not tossing anything (I get the dice parallel but let's be clear).